### PR TITLE
Remove usage of `Config` in favor of domain objects

### PR DESCRIPTION
**Pull request by jcazevedo**
This PR removes some usage of typesafe's `Config` in favor of domain objects, by depending on [PureConfig](https://github.com/melrief/pureconfig) for `Config` deserialization.

### CODE REVIEW

File: src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala [hunk 1]
```diff
@@ -43,12 +45,12 @@ trait TrackingListener extends RabbitMQUtilAsync {
   /**
    * The name of the AMQP exchange from which the listener will consume events.
    */
-  lazy val exchangeName = trackingConfig.getString("exchange-name")
+  lazy val exchangeName = trackingConfig.exchangeName
```
Comment (ruippeixotog) [inline src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala]: Do we still need these `lazy vals` for each extracted setting? They sure can be here, but they aren't that needed anymore, as accessing a field in a class is pretty fast and there is no parsing anymore.

File: README.md [hunk 2]
```diff
@@ -98,23 +99,19 @@ credentials `guest`/`guest`) and using the default RPC configuration:
 scala> import eu.shiftforward.adstax.productfeeder.rpc._
 import eu.shiftforward.adstax.productfeeder.rpc._
 
-scala> import com.typesafe.config._
-import com.typesafe.config._
+scala> import eu.shiftforward.adstax.config._
```
Comment (jcazevedo) [inline README.md]: Yep. Fixed in d23710b.

File: README.md [hunk 3]
```diff
@@ -98,23 +99,19 @@ credentials `guest`/`guest`) and using the default RPC configuration:
...
+import eu.shiftforward.adstax.config
 
 scala> import akka.actor._
 import akka.actor._
 
+scala> import scala.concurrent.duration._
```
Comment (jcazevedo) [inline README.md]: Yep. Fixed in d23710b.

File: README.md [hunk 4]
```diff
@@ -338,17 +334,15 @@ class ListenerActor extends Actor {
...
-    exchange-name = "sf.data"
-    routing-key = "http.request.json"
-  }""")
+  lazy val rmqConfig = config.RabbitMQ(
+    "mq.sample-adstax-instance.dev.adstax.io",
+    5672,
+    "guest",
+    "guest",
+    5.minutes)
+  lazy val trackingConfig = RabbitMQ.BindingConfig(
```
Comment (jcazevedo) [inline README.md]: It doesn't. I forgot to scope it under the `config` package. Fixed in d23710b.

File: src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala (hunk 1 above)
Comment (jcazevedo) [inline src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala]: We don't. Fixed in 9154fb1.

**Comment by ruippeixotog** [comment]
LGTM!

**Pull request merged**